fe1-web/network/NetworkConnection.ts

```diff
@@ -23,65 +24,83 @@ interface PendingResponse {
+        }
       }
 
-      if (response.id !== UNDEFINED_ID) {
-        const pendingResponse = this.payloadPending.get(response.id);
-        if (pendingResponse === undefined) {
-          throw new NetworkError(`Received a response which id = ${response.id} does not match any pending requests`);
-        }
+      if (parsedMessage instanceof JsonRpcResponse) {
+        // if we have a response
+        if (parsedMessage.id !== UNDEFINED_ID) {
+          const pendingResponse = this.payloadPending.get(parsedMessage.id);
+          if (pendingResponse === undefined) {
+            throw new NetworkError(`Received a response which id = ${parsedMessage.id} does not match any pending requests`);
+          }
 
-        // a response was received, we clear the timeout and the pending query from the pending map
-        clearTimeout(pendingResponse.timeoutId);
-        this.payloadPending.delete(response.id);
-
-        // reject the promise if the response is negative
-        if (response.isPositiveResponse()) {
-          pendingResponse.resolvePromise(response);
-          try {
-            this.onMessageHandler(response);
-          } catch (e) {
-            console.error('Exception encountered when giving response : ', response, ' to its handler.\n', e);
+          // a response was received, clear the timeout and the pending query from the pending map
+          clearTimeout(pendingResponse.timeoutId);
+          this.payloadPending.delete(parsedMessage.id);
+
+          // reject the promise if the response is negative
+          if (parsedMessage.isPositiveResponse()) {
+            pendingResponse.resolvePromise(parsedMessage);
+            try {
+              this.onMessageHandler(parsedMessage);
+            } catch (e) {
+              console.error('Exception encountered when giving response : ', parsedMessage, ' to its handler.\n', e);
+            }
+          } else {
+            // Note : impossible to have an undefined error from now on due to isPositiveResponse()
+            pendingResponse.rejectPromise(
+              'A negative network error was received:\n'
+              + `\t- error code : ${parsedMessage.error?.code}`
+              + `\t- description : ${parsedMessage.error?.description}`,
+            );
           }
-        } else {
-          // Note : impossible to have an undefined error at this stage due to isPositiveResponse()
-          pendingResponse.rejectPromise(
-            'A negative network error was received:\n'
-            // @ts-ignore
-            + `\t- error code : ${response.error.code}`
-            // @ts-ignore
-            + `\t- description : ${response.error.description}`,
-          );
         }
+      } else {
+        // if we have a request which needs to be forwarded
+
+        // Note to students : here should be the logic regarding witness message forwarding
+        getNetworkManager().sendPayload(parsedMessage); // FIXME ignoring promise for now
```

Comment:
here we should already be parsing Broadcast messages.. 😉

Reply:
Fixed it.

Note : added 2 eslint-disable next to it for good reasons this time ^^